File: salleManagementFrontend/frontend/src/services/ligneVenteService.js
import api from "./api";

export function addLine(saleId, data) {
  return api.post(`/ligne-ventes/${saleId}`, data);
}

export function updateLine(lineId, data) {
  return api.put(`/ligne-ventes/${lineId}`, data);
}

export function deleteLine(lineId) {
  return api.delete(`/ligne-ventes/${lineId}`);
}

File: backend/sallesMangement/src/main/java/com/projetee/sallesmangement/service/LigneVenteService.java
package com.projetee.sallesmangement.service;

import com.projetee.sallesmangement.dto.lignevente.LigneVenteRequest;
import com.projetee.sallesmangement.dto.lignevente.LigneVenteResponse;
import org.springframework.data.domain.Page;

public interface LigneVenteService {

    LigneVenteResponse addLine(Long saleId, LigneVenteRequest request);

    LigneVenteResponse updateLine(Long lineId, LigneVenteRequest request);

    void deleteLine(Long lineId);

    LigneVenteResponse get(Long id);

    Page<LigneVenteResponse> getPaginated(int page, int size);

}
